src/protocol/attribution/aggregate_credit.rs

```diff
@@ -100,33 +261,35 @@ async fn bit_decompose_breakdown_key<F: Field>(
 }
 
 /// Sort the input by `aggregation_bit` first, then by `breakdown_key`
-#[allow(dead_code)]
 async fn sort_by_aggregation_bit_and_breakdown_key<F: Field>(
     ctx: SemiHonestContext<'_, F>,
     input: &[CappedCreditsWithAggregationBit<F>],
 ) -> Result<Vec<CappedCreditsWithAggregationBit<F>>, Error> {
-    // Sort by aggregation_bit
-    let sorted_by_aggregation_bit = sort_by_aggregation_bit(ctx.clone(), input).await?;
+    // Prepend the aggregation_bit bits with breakdown_key bits. This results
+    // in sorting the sidecar by aggregation_bit first, and then by breakdown_key.
+
+    let mut permutation_keys = vec![input
+        .iter()
+        .map(|x| x.aggregation_bit.clone())
+        .collect::<Vec<_>>()];
 
-    // Next, sort by breakdown_key
     // TODO: Change breakdown_keys to use XorReplicated to avoid bit-decomposition calls
-    let breakdown_keys = transpose(
-        &bit_decompose_breakdown_key(
-            ctx.narrow(&Step::BitDecomposeBreakdownKey),
-            &sorted_by_aggregation_bit,
-        )
-        .await?,
+    let mut breakdown_keys = transpose(
+        &bit_decompose_breakdown_key(ctx.narrow(&Step::BitDecomposeBreakdownKey), input).await?,
```

Comment:
[Comment from #326 
](https://github.com/private-attribution/ipa/pull/326#discussion_r1041632591)

> We can probably optimize this right? This is a function of the maximum value of "breakdown key", which should be a known constant. I'd suggest making this an argument to the function.

I didn't understand what you meant by "the maximum value of "breakdown key"? Bit-decomposition is applied to all breakdown_key shares in the input rows.

Since this function is called only once per aggregation protocol call, I left it as it is for now. I'll optimize in a later diff if needed.

Reply:
I left a comment explaining it below. See my comment below next to the line:
```
F::Integer::BITS + 1, // breakdown_keys bits + aggregation_bit bit
```